### ISSUE TITLE
Q-Q Plot

### ISSUE THREAD

**Issue by aimalz** [aimalz/qp#42]
I was thinking it might be useful to implement a quantile-quantile plot option in addition to calculating the RMS and KLD to compare `PDF` objects.

Edit: This will probably also merit another explanatory notebook like `kld.ipynb`.

**Comment by drphilmarshall** [comment]
Will this help us optimize the algorithm for choosing the quantiles? I
still wonder whether regular separations could be improved upon.

**Comment by aimalz** [comment]
I'm not sure if it will help with #35, but I can imagine a way to use the q-q values to choose the optimal quantile spacing.  What if we ran an optimizer over the quantiles to minimize the L1 or L2 norm between the true PDF and the quantile PDF?  A corollary to this idea is that the bin endpoints could be optimized using the q-q between the true PDF and the binned PDF, which would be guaranteed to be just as good of an approximation as optimized quantiles.  (The biggest remaining advantage to the quantiles would then be that the approximated PDF information could be stored more compactly. EDIT: Actually, there's no reason for me to think that -- the two options would be more or less indistinguishable under an optimization of this type. . .)  I'll try it out!

**Comment by drphilmarshall** [comment]
Optimizing the quantiles is a great idea! Only connect, eh? :-)

You could try optimizing the MSE/L2 norm and then compare with optimizing
the KLD, and look for differences in both quality of result and
computational efficiency.

BTW I think you will want to do these tests on a large number of mock 1D
PDFs, so you can "average over an effective ensemble of photo-z datasets".
The goal is to find a _robust_ PDF approximation scheme - ie one that works
in "all cases"... Weren't we going to generate mock photo-z PDFs using some
mixture of Gaussian scheme? We may even have an open issue...

**Comment by aimalz** [comment]
Indeed, #36 includes mention of nontrivial PDFs, which I envisioned being mixtures of scipy.stats.rv_continuous objects.  However, I haven't yet thought about deploying `qp` on a "survey" of mock 1D PDFs.  Perhaps that should be a separate issue.

**Comment by drphilmarshall** [comment]
Yes, I think you're right - separate out the mixture of RVs capability from
the simulating a survey project notebook.
